File: completable-reactor-graph/src/main/java/ru/fix/completable/reactor/graph/RoutingMerger.java
package ru.fix.completable.reactor.graph;

@FunctionalInterface
public interface RoutingMerger<Payload, HandlerResult> {
    Enum merge(Payload payload, HandlerResult handlerResult);
}
//TODO replace Enum with Any?
//TODO move GL graph API to separate package
//TODO create scala API proposal

File: completable-reactor-graph/src/main/kotlin/ru/fix/completable/reactor/graph/RoutingMerger.kt
package ru.fix.completable.reactor.graph

@FunctionalInterface
interface RoutingMerger<Payload, HandlerResult> {
    fun merge(payload: Payload, handlerResult: HandlerResult): Enum<*>
}
//TODO replace Enum with Any?
//TODO move GL graph API to separate package
//TODO create scala API proposal
